qiskit_aer/primitives/sampler_v2.py

```diff
@@ -0,0 +1,261 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Sampler V2 class.
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Iterable
+import warnings
+
+import numpy as np
+from numpy.typing import NDArray
+
+from qiskit import ClassicalRegister, QiskitError, QuantumCircuit
+from qiskit.circuit import ControlFlowOp
+
+from qiskit.primitives.base import BaseSamplerV2
+from qiskit.primitives.base.validation import _has_measure
+from qiskit.primitives.containers import (
+    BitArray,
+    PrimitiveResult,
+    PubResult,
+    SamplerPubLike,
+    make_data_bin,
+)
+from qiskit.primitives.containers.sampler_pub import SamplerPub
+from qiskit.primitives.containers.bit_array import _min_num_bytes
+from qiskit.primitives.primitive_job import PrimitiveJob
+
+from qiskit_aer import AerSimulator
+
+
+@dataclass
+class _MeasureInfo:
+    creg_name: str
+    num_bits: int
+    num_bytes: int
+    qreg_indices: list[int]
+
+
+class SamplerV2(BaseSamplerV2):
+    """
+    Aer implementation of SamplerV2 class.
+
+    Each tuple of ``(circuit, <optional> parameter values, <optional> shots)``, called a sampler
+    primitive unified bloc (PUB), produces its own array-valued result. The :meth:`~run` method can
+    be given many pubs at once.
+    """
+
+    def __init__(
+        self,
+        *,
+        backend: AerSimulator = None,
```

Comment:
Could you consider the options class and backend_options adopted by estimator for consistency. I am not sure which is better. Perhaps the Estimator side should be changed to match this specification.

Reply:
Note that this option class comes from BackendSampler/Estimator.